### PR TITLE
Mdm 338: document JSON options format

### PR DESCRIPTION
**Pull request by dmcassel**

### CODE REVIEW

File: _pages/docs/matching-options.md [hunk 1]
```diff
@@ -74,6 +74,90 @@ Here's an example of match configuration options.
...
+        { "namespace": "", "localname": "AddressPrivateMailboxText", "name": "addr1" },
+        { "namespace": "", "localname": "LocationCity", "name": "city" },
+        { "namespace": "", "localname": "LocationState", "name": "state" },
+        { "namespace": "", "localname": "LocationPostalCode", "name": "zip" }
+      ]
+    },
+    "algorithms": {
+      "algorithm": [
+        { "name": "std-reduce", "function": "standard-reduction" },
+        { "name": "std-reduce-query", "function": "standard-reduction-query" },
```
Comment (paxtonhare) [inline _pages/docs/matching-options.md]: remove the std-reduce-query one because it's not being used.

File: _pages/docs/matching-options.md [hunk 2]
```diff
@@ -74,6 +74,90 @@ Here's an example of match configuration options.
...
+          { "property-name": "first-name", "weight": "12" },
+          { "property-name": "addr1", "weight": "5" },
+          { "property-name": "city", "weight": "3" },
+          { "property-name": "state", "weight": "1" },
+          { "property-name": "zip", "weight": "3" }
+        ],
+        "expand": [
+          {
+            "property-name": "first-name",
+            "algorithm-ref": "thesaurus",
```
Comment (paxtonhare) [inline _pages/docs/matching-options.md]: probably too late for this comment... should we do camelCase?

File: _pages/docs/merging-options.md [hunk 3]
```diff
@@ -86,6 +87,98 @@ Here's an example of merge configuration options.
...
+      },
+      "custom": [
+        { "name": "name", "function": "name", "at": "" },
+        { "name": "address", "function": "address", "at": "" }
+      ]
+    },
+    "merging": [
+      {
+        "propertyName": "ssn",
+        "algorithmRef": "user-defined",
```
Comment (paxtonhare) [inline _pages/docs/merging-options.md]: remove this unused algorithmRef

File: _pages/docs/matching-options.md (hunk 2 above)
Comment (paxtonhare) [reply _pages/docs/matching-options.md]: actually I see camel case in some places. weird.

File: _pages/docs/matching-options.md [hunk 4]
```diff
@@ -74,6 +74,90 @@ Here's an example of match configuration options.
...
+        { "namespace": "", "localname": "PersonGivenName", "name": "first-name" },
+        { "namespace": "", "localname": "PersonSurName", "name": "last-name" },
+        { "namespace": "", "localname": "AddressPrivateMailboxText", "name": "addr1" },
+        { "namespace": "", "localname": "LocationCity", "name": "city" },
+        { "namespace": "", "localname": "LocationState", "name": "state" },
+        { "namespace": "", "localname": "LocationPostalCode", "name": "zip" }
+      ]
+    },
+    "algorithms": {
+      "algorithm": [
```
Comment (paxtonhare) [inline _pages/docs/matching-options.md]: this feels lame. I'd expect algorithms: []

Comment (dmcassel) [reply _pages/docs/matching-options.md]: for merging, the algorithms object has keys `custom` (array) and `stdAlgorithm`. I'm inclined to leave `algorithms` as an object and to change `algorithm` to `custom` to be consistent. Thoughts?

File: _pages/docs/merging-options.md (hunk 3 above)
Comment (dmcassel) [reply _pages/docs/merging-options.md]: removed

File: _pages/docs/matching-options.md (hunk 1 above)
Comment (dmcassel) [reply _pages/docs/matching-options.md]: removed

File: _pages/docs/matching-options.md (hunk 2 above)
Comment (dmcassel) [reply _pages/docs/matching-options.md]: I agree, should be camel case

File: _pages/docs/matching-options.md (hunk 4 above)
Comment (paxtonhare) [reply _pages/docs/matching-options.md]: custom works for me

**Pull request merged**